docs/dev/nomenclature.md

```diff
@@ -0,0 +1,13 @@
+## Nomenclature
+
```

Comment:
Can we write a short introduction here?

Something like, "This page describes naming rules and conventions that exist throughout cirq.  These rules are important so that we maintain a consistent interface that is easy to use...."

Reply:
Done